2007-07-15  Andrew Haley  <aph@redhat.com>

	* unwind-sjlj.c (_Unwind_GetIPInfo): Check for context->fc != NULL
	before looking in the context.

	* configure.host: 
	arm*-linux-gnu -> arm*-linux*.

--- gcc/unwind-sjlj.c	(revision 126657)
+++ gcc/unwind-sjlj.c	(revision 126659)
@@ -222,7 +222,10 @@ _Unwind_Ptr
 _Unwind_GetIPInfo (struct _Unwind_Context *context, int *ip_before_insn)
 {
   *ip_before_insn = 0;
-  return context->fc->call_site + 1;
+  if (context->fc != NULL)
+    return context->fc->call_site + 1;
+  else
+    return 0;
 }
 
 /* Set the return landing pad index in CONTEXT.  */
--- libjava/configure.host	(revision 126657)
+++ libjava/configure.host	(revision 126659)
@@ -82,7 +82,7 @@ case "${host}" in
 	enable_getenv_properties_default=no
 	enable_main_args_default=no
 	;;
-  arm*-linux-gnu)
+  arm*-linux*)
 	libgcj_interpreter=yes
 	sysdeps_dir=arm
 	;;
